.github/workflows/feature-branch-auto-merge.yml

```diff
@@ -0,0 +1,34 @@
+name: Feature branch auto merge
+
+permissions: read-all
+
+on:
+  push:
+    branches:
+      - main  # Set a branch to deploy
+      
+env:
+
+  MY_REPO: https://GoogleCloudPlatform:${{secrets.GITHUB_TOKEN}}@github.com/GoogleCloudPlatform/magic-modules.git
+
+  MY_BRANCH: feature-branch-release-5.0.0
+
+  MAIN_REPO: https://github.com/GoogleCloudPlatform/magic-modules.git
+
+  MAIN_BRANCH: main
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    
+    permissions:
+      contents: write
+
+    steps:
+    - name: Merge with main
+      run: |
+        git clone ${{env.MY_REPO}} -b ${{env.MY_BRANCH}} tmp
```

Comment:
I would expect the action to already have the working directory be a clone of the main branch; would it be possible to check out the feature branch directly and then merge the local main branch instead of cloning and pulling? Not a huge concern, just seems like it might be slightly simpler.